### ISSUE TITLE
[tx] (t1r) /BlueScale too large/small (ignored)

### ISSUE THREAD

**Issue by BoldMonday** [adobe-type-tools/afdko#767]
Using AFDKO 2.8.5 and `makeotf` to compile a font I get the following report in Terminal:
```
tx: --- font.pfa
tx: (t1r) /BlueScale too large/small (ignored)
tx: --- font.pfa
tx: (t1r) /BlueScale too large/small (ignored)
Built release mode font 'IBMPlexSans.otf' Revision 3.001
```
BlueScale value in the source font.pfa is 0.039625 on a standard UPM of 1000.
Never had any issues with this before.

Is this a error on our part, or in tx?

**Comment by frankrolf** [comment]
Curious about this one too. I am getting it sporadically, but haven’t found a scenario to reproduce.

**Comment by kenlunde** [comment]
I am not sure whether this helps, but I usually encounter this warning when the stem values in a stem-value pair are too close. Simply indicating the hinting parameters would be helpful.

**Comment by BoldMonday** [comment]
Thanks. Here they are @kenlunde:
```
PS Type 1 Vstems: 80, 68, 86
PS Type 1 Hstems: 74, 68
```

**Comment by kenlunde** [comment]
Alignment zones, too, please, meaning /BlueValues and /OtherBlues.

**Comment by BoldMonday** [comment]
Blue Zones all have a width of 12 units, except for one zone of 6 units.
Is that last one the culprit, perhaps?

**Comment by kenlunde** [comment]
Yes, the six-unit would definitely be the culprit, at least in my experience. As an aside, the stems should be in ascending order.

**Comment by kenlunde** [comment]
I generally set stems to be at least 10 units apart. If you need to pick between two that are close, always go with the smaller value.

**Comment by frankrolf** [comment]
The stems should be in ascending order following the most dominant stem (`StdVW`, `StdHW`), which is defined by its first position in the array, but can have a larger value than the smallest stem of the overall stem bouquet (`StemSnapV`, `StemSnapH`).

**Comment by miguelsousa** [comment]
@BoldMonday the value of `BlueScale` needs to be set according to the **alignment zones** the font has. To pick an appropriate value use the formula shown in this presentation https://vimeo.com/38364880

**Comment by frankrolf** [comment]
According to the formula given in the presentation (`3 / (4 * maximum_zone_size)`), the blueScale value for your project should be `0.0625`.
I’m quite familiar with that formula these days since it comes up in FDK workshops everywhere! :-)

**Comment by frankrolf** [comment]
Notwithstanding, I’m still curious what kinds of values trigger the “too large/small” warning.

**Comment by BoldMonday** [comment]
Thanks for the valuable feedback, all.

@miguelsousa Following your logic it means that all Blue Zones in a font (and typically a font family too) need to have the exact same distance?

**Comment by miguelsousa** [comment]
> I’m still curious what kinds of values trigger the “too large/small” warning.

The ones outside this range:

1 / (2 * MaxZoneSize) <= BlueScale < 1 / MaxZoneSize

**Comment by miguelsousa** [comment]
> Following your logic it means that all Blue Zones in a font (and typically a font family too) need to have the exact same distance?

No, they don't. It just means that the value of BlueScale is conditioned by the height of the largest zone.

**Comment by miguelsousa** [comment]
> Never had any issues with this before.
> Is this a error on our part, or in tx?

The `tx` messages began being exposed after this change 5f6f5752965263a2a712b126802cc255d737411f in `makeotf`, which shipped with **afdko v2.8.5**. 
Before this change, the messages simply got discarded (because of the use of `2>&1` in the call to tx) and never presented to the user.

The error is in the font files.
